### DISCUSSION TITLE
Detect datepicker change event

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#11919] [Help]
### Package

Panel builder

### Package Version

v3.2.51

### How can we help you?

How can I get the datepicker change event? `extraInputAttributes`, `extraAlpineAttributes`, `extraAttributes` none of these options add an `@change` attribute to the input. My use case is to open the select field immediately after selecting the datepicker.

**Comment by leandrocfe** [comment]
`afterStateUpdated()` ?

**Reply by ArtMin96** [reply]
Thank you. I don't want to send a request to the server to solve such a problem.

**Reply by ArtMin96** [reply]
date-time-picker.blade.php [**Line: 210**]

```html
x-on:click="dayIsDisabled(day) || selectDate(day) || $dispatch('date-selected', {selected: true})"
```

```php
Select::make('time')
    ->native(false)
    ->extraInputAttributes([
        '@date-selected.window' => 'console.log($refs.input, $event.details.selected)',
    ])
```

In this way I was able to get what I needed, but now I don't know how to focus select (`choices`).

**Reply by ArtMin96** [reply]
If I solve the problem with `afterStateUpdated`, then how will I show the dropdown?

**Reply by ArtMin96** [reply]
I almost got it working. The only problem is that I cannot specify which select will open if there is more than 1 select field.

![7](https://github.com/filamentphp/filament/assets/29732308/e3f67dcf-a078-4d0e-8329-3e746813abd6)


```php
->extraInputAttributes([
    '@date-selected.window' => 'select.showDropdown()',
])
```
